### PR TITLE
fix: use status instead of event for pdb checking

### LINKED ISSUE #476: [Question]: should not use the history event for the pdb checking
### Checklist

- [X] I've searched for similar issues and couldn't find anything matching
- [X] I've included steps to reproduce the behavior

### Affected Components

- [X] K8sGPT (CLI)
- [ ] K8sGPT Operator

### K8sGPT Version

k8sgpt: 0.3.6 (9c0efe6), built at: unknown

### Kubernetes Version

_No response_

### Host OS and its Version

Mac OS 12.6.6

### Steps to reproduce

1. Run `$ k8sgpt analyze`
2. 

### Expected behaviour

No error report for the PDB(`PodDisruptionBudget`).

### Actual behaviour

```console
MacBook-Pro:~ jianzhang$ k8sgpt analyze -n openshift-operator-lifecycle-manager
AI Provider: openai

0 openshift-operator-lifecycle-manager/packageserver-pdb(packageserver-pdb)
- Error: No matching pods found, expected label app=packageserver 

MacBook-Pro:~ jianzhang$ k8sgpt analyze 
AI Provider: openai

0 openshift-kube-apiserver/kube-apiserver-guard-pdb(kube-apiserver-guard-pdb)
- Error: No matching pods found, expected label app=guard

1 openshift-monitoring/prometheus-operator-admission-webhook(prometheus-operator-admission-webhook)
- Error: No matching pods found, expected label app.kubernetes.io/name=prometheus-operator-admission-webhook
- Error: No matching pods found, expected label app.kubernetes.io/part-of=openshift-monitoring

2 openshift-oauth-apiserver/oauth-apiserver-pdb(oauth-apiserver-pdb)
- Error: No matching pods found, expected label apiserver=true
- Error: No matching pods found, expected label app=openshift-oauth-apiserver

3 openshift-apiserver/openshift-apiserver-pdb(openshift-apiserver-pdb)
- Error: No matching pods found, expected label apiserver=true
- Error: No matching pods found, expected label app=openshift-apiserver-a

4 openshift-cloud-credential-operator/pod-identity-webhook(pod-identity-webhook)
- Error: No matching pods found, expected label app=pod-identity-webhook

5 openshift-cluster-csi-drivers/aws-ebs-csi-driver-controller-pdb(aws-ebs-csi-driver-controller-pdb)
- Error: No matching pods found, expected label app=aws-ebs-csi-driver-controller

6 openshift-etcd/etcd-guard-pdb(etcd-guard-pdb)
- Error: No matching pods found, expected label app=guard

7 openshift-kube-controller-manager/kube-controller-manager-guard-pdb(kube-controller-manager-guard-pdb)
- Error: No matching pods found, expected label app=guard

8 openshift-kube-scheduler/openshift-kube-scheduler-guard-pdb(openshift-kube-scheduler-guard-pdb)
- Error: No matching pods found, expected label app=guard

9 openshift-operator-lifecycle-manager/packageserver-pdb(packageserver-pdb)
- Error: No matching pods found, expected label app=packageserver
```

### Additional Information

The PDB works as expected, and it already found the matched pods, but the history event is preserved. We shouldn't use the `event` to judge if the pdb works well.
```console 
MacBook-Pro:~ jianzhang$ oc get pods --show-labels|grep packageserver
packageserver-69cc8b8cdd-jshp4            1/1     Running     0          106m    app=packageserver,pod-template-hash=69cc8b8cdd
packageserver-69cc8b8cdd-nszzz            1/1     Running     0          106m    app=packageserver,pod-template-hash=69cc8b8cdd

MacBook-Pro:~ jianzhang$ oc describe pdb packageserver-pdb
Name:             packageserver-pdb
Namespace:        openshift-operator-lifecycle-manager
Max unavailable:  1
Selector:         app=packageserver
Status:
    Allowed disruptions:  1
    Current:              2
    Desired:              1
    Total:                2
Events:
  Type    Reason  Age                  From               Message
  ----    ------  ----                 ----               -------
  Normal  NoPods  139m (x2 over 139m)  controllermanager  No matching pods found

MacBook-Pro:~ jianzhang$ oc get event|grep packageserver-pdb
115m        Normal    NoPods                poddisruptionbudget/packageserver-pdb          No matching pods found
```

### PR DESCRIPTION
**Pull request by jianzhangbjz**
Closes #476 

## 📑 Description
The PDB event is a historical record, it's not a good choice to judge `pdb` status by based on it. So, I used the `status` instead of `event` in this PR.

## ✅ Checks

- [ ] My pull request adheres to the code style of this project
- [ ] My code requires changes to the documentation
- [ ] I have updated the documentation as required
- [ ] All the tests have passed

### CODE REVIEW

File: pkg/analyzer/pdb.go [hunk 1]
```diff
@@ -50,48 +50,28 @@ func (PdbAnalyzer) Analyze(a common.Analyzer) ([]common.Result, error) {
...
+			if pdb.Spec.MaxUnavailable != nil {
+				doc = apiDoc.GetApiDocV2("spec.maxUnavailable")
+			}
+			if pdb.Spec.MinAvailable != nil {
+				doc = apiDoc.GetApiDocV2("spec.minAvailable")
+			}
+			for k, v := range pdb.Spec.Selector.MatchLabels {
 				failures = append(failures, common.Failure{
-					Text:          fmt.Sprintf("%s, selector is nil", evt.Message),
+					Text:          fmt.Sprintf("%s, expected pdb pod lable %s=%s", pdb.Status.Conditions[0].Reason, k, v),
```
Comment (AlexsJones) [inline pkg/analyzer/pdb.go:63]:
```suggestion
Text:          fmt.Sprintf("%s, expected pdb pod label %s=%s", pdb.Status.Conditions[0].Reason, k, v),
```

**Review by arbreezy — CHANGES_REQUESTED** [review]
Update your commit message as well, to conform with release-please

File: pkg/analyzer/pdb.go (hunk 1 above)
Comment (jianzhangbjz) [reply pkg/analyzer/pdb.go]: Updated it, thanks!

**Review by AlexsJones — APPROVED** [review]
LGTM, but as @arbreezy says please update commit message

**Comment by jianzhangbjz** [comment]
>  please update commit message

Hi @AlexsJones @arbreezy , thanks for your review. I will squash those two commits, but, sorry, what commit message I should add? Thanks!

**Comment by jianzhangbjz** [comment]
Hi @AlexsJones @AlexsJones , I updated the commit message, please let me know if any thing else is needed, thanks!
```console
    fix: use the `status` for pdb checking
    
    The PDB event is a historical record, it's not a good choice to judge the pdb latest status based on it.
    So, use the `stataus` instead of `event` to check it.
    
    Closes: #476
    
    Signed-off-by: Jian Zhang <jiazha@redhat.com>
```

**Review by arbreezy — APPROVED** [review]

**Comment by jianzhangbjz** [comment]
Hi @AlexsJones @arbreezy, seems like the merging is blocked by `The base branch restricts merging to authorized users.`. Could you help have a look? Thanks!
image

**Pull request merged**